backend/generating/SupervisionTemplate.java

```diff
@@ -0,0 +1,69 @@
+package generation;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class SupervisionTemplate{
+
+    private SupervisionInfo supervisionInfo;
+    private SupervisionQuestions questions;

```

Comment:
What is SupervisionQuestions? I don't think this is included in this pull request?